File: backend/app/schemas/repository.py
from datetime import datetime
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field
from app.schemas.repository_automation import RepositoryAutomationResponse


class RepositoryBase(BaseModel):
    github_repo_id: str
    name: str = Field(..., max_length=255)
    full_name: str = Field(..., max_length=510)
    default_branch: str = Field(default="main", max_length=100)
    is_private: bool = False
    html_url: Optional[str] = None
    description: Optional[str] = None


class RepositoryCreate(RepositoryBase):
    user_id: UUID


class RepositoryUpdate(BaseModel):
    default_branch: Optional[str] = None
    is_private: Optional[bool] = None
    html_url: Optional[str] = None
    description: Optional[str] = None


class RepositoryResponse(RepositoryBase):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    automation: Optional[RepositoryAutomationResponse] = None
    created_at: datetime
    updated_at: datetime


class RepositoryDetailResponse(RepositoryResponse):
    execution_count: int = 0

File: frontend/src/types/repository.ts
export type AutomationStatus = 'ACTIVE' | 'INACTIVE';

export interface RepositoryAutomation {
  id: string;
  repository_id: string;
  status: AutomationStatus;
  target_branch: string;
  doc_paths: string[];
  auto_commit: boolean;
  create_pull_request: boolean;
  pr_target_branch: string;
  last_activated_at?: string;
  last_deactivated_at?: string;
  created_at: string;
  updated_at: string;
}

export interface Repository {
  id: string;
  user_id?: string;
  github_repo_id: string;
  name: string;

  full_name: string;
  default_branch: string;
  is_private: boolean;
  html_url?: string;
  description?: string;
  language?: string;
  automation?: RepositoryAutomation;
  execution_count?: number;
  created_at: string;
  updated_at: string;
}

export interface RepositoryUpdatePayload {
  default_branch?: string;
  is_private?: boolean;
  description?: string;
}

export interface AutomationUpdatePayload {
  target_branch?: string;
  doc_paths?: string[];
  auto_commit?: boolean;
  create_pull_request?: boolean;
  pr_target_branch?: string;
}

export interface AutomationToggleResponse {
  repository_id: string;
  status: AutomationStatus;
  message: string;
  updated_at: string;
}
